### ISSUE TITLE
XCode build fails to produce .pdx, command line build works

### ISSUE THREAD

**Issue by cbackas42** [finnvoor/PlaydateKit#14]
Hello, and thanks for the great work on this!

I followed the setup instructions, and while `swift package pdc` indeed produces a pdx under the .build folder, building it in Xcode does not. Command-R completes without errors, but it launched the Playdate simulator without any file loaded because there is no PDX in the TARGET_BUILD_DIR directory.

Although I've gone over the steps a few times it's entirely possible I've missed something, but I'm not able to spot it.

**Comment by finnvoor** [comment]
If the command line works but Xcode doesn't then it seems to be a problem with the xcscheme settings.  I just went through the steps again myself and it worked fine, but maybe there's a mismatch between our environments or something.

If you followed the *If you will be running your game from Xcode, you will need to modify the default xcschemes.* section of "Building the Template" then your scheme should look like this:

![CleanShot 2024-03-23 at 10 05 54@2x](https://github.com/finnvoor/PlaydateKit/assets/8284016/efd1a0f9-49e7-4b50-9f70-b998e8456d8c)
![CleanShot 2024-03-23 at 10 06 05@2x](https://github.com/finnvoor/PlaydateKit/assets/8284016/b6d60be7-9fc5-44d8-9182-f3ec72683360)
![CleanShot 2024-03-23 at 10 06 12@2x](https://github.com/finnvoor/PlaydateKit/assets/8284016/daf3f6b0-f069-4916-b527-2b84338e3965)

If your setup looks the same then I'll investigate a bit more and try to figure it out!

**Comment by finnvoor** [comment]
Another thing you can try is to just clone the template and skip the renaming steps, and try to run the template from Xcode as is, that should 100% work unless there's something I'm not seeing.

**Comment by finnvoor** [comment]
I've updated the template to build in the run phase now so you will see any errors in the Xcode console.  It also doesn't require as many env vars, so I think the updated template will fix your issue. Please reopen if you are still having trouble!

**Comment by cbackas42** [comment]
Thanks for responding quickly! I tried your first suggestion, and tried building the unmodified PlaydateKit template project. The results were exactly the same, worked from the CLI but not from Xcode, the Simulator launched with no file loaded. Just like in my own project, this was the entire contents of TARGET_BUILD_DIR:
Mar 23 11:35 ./
Mar 23 11:35 ../
Mar 23 11:35 PackageFrameworks/
Mar 23 11:35 PlaydateKit.o
Mar 23 11:35 PlaydateKit.swiftmodule/
Mar 23 11:35 PlaydateKitTemplate.o
Mar 23 11:35 PlaydateKitTemplate.swiftmodule/

Then I pulled your latest changes. Same ultimate result, it still works from the CLI but not from Xcode; however, the Xcode behavior has changed as it now reports errors and doesn't launch the simulator. These are the console outputs from Command-R:
No matching processes belonging to you were found
Logging Error: Failed to initialize logging system. Log messages may be missing. If this issue persists, try setting IDEPreferLogStreaming=YES in the active scheme actions environment variables.
found Swift toolchain: org.swift.59202403191a
found Playdate SDK
[2K
[31m[1merror: [0mPlugin does not have access to a tool named ‘arm-none-eabi-gcc’
Program ended with exit code: 1
